include/dr/detail/communicator.hpp

```diff
@@ -264,12 +273,23 @@ class rma_window {
 
   void put(const void *src, std::size_t size, std::size_t rank,
            std::size_t disp) const {
-    DRLOG("MPI comm put:: ({}:{}:{})", rank, disp, size);
+    DRLOG("MPI comm put:: ({}:{}:{})", rank, disp, (int)size);
     MPI_Request request;
+
+#if (MPI_VERSION >= 4) ||                                                      \
```

Comment:
to not repeat long condition please create: `#define MPI_SUPPORTS_RGET_C ...` and use in both places